types/concurrency-limiter/limiter.d.ts

```diff
@@ -0,0 +1,46 @@
+/**
+  The concurrency limiter.
+
+  It is instantiated providing the maximum number of async tasks that are
+  allowed to run concurrently.
+*/
+declare class Limiter {
+    constructor(limit: number);
+
+    private _limit: number;
+    private _length: number;
+    private _queue: Array<() => void>;
+
```

Comment:
those are not needed IMO (unless used as indexed access types), users will get errors anyway when using non-private/non-protected fields, etc.